DownFall/Scenes/GameScene.swift

```diff
@@ -159,6 +170,27 @@ class GameScene: SKScene {
         Dispatch.shared.reset()
         print("deiniting")
     }
+    
+    /// Return the current tiles to save
+    func currentTilesToSave() -> [[Tile]] {
+        print("grabbing tiles from the board")
+        return board?.tiles ?? []
+    }
+    
+    func saveAllState() {
+        guard let board = self.board,
+              let playerIndex = tileIndices(of: .player(.zero), in: board.tiles).first,
+              case let TileType.player(data) = board.tiles[playerIndex].type,
+              let levelGoalTracking = renderer?.goalTrackingData()
+              else {
+            fatalError("we lack enough information to save the game")
+        }
+        
+        self.gameSceneDelegate?.saveStateOnWin(self,
```

Comment:
Let's rename the delegate method since we are using it at different times then just during a win